osf_tests/metrics/reporters/test_institutional_users_reporter.py

```diff
@@ -159,6 +165,15 @@ def __post_init__(self):
             ),
         )
         self._add_affiliations(self._generate_counted_objects())
+        node_logs = self.user.logs.order_by('-created')
+        preprint_logs = self.user.preprint_logs.order_by('-created')
+
+        dates = filter(bool, [
+            node_logs.values_list('created', flat=True).first(),
+            preprint_logs.values_list('created', flat=True).first(),
+        ])
+
+        self.month_last_active = max(dates, default=None)
```

Comment:
oh, small point of "names shouldn't lie" (not to mention that type annotations shouldn't lie):

if leaving names (and annotations) as they are, should cast this to `YearMonth` here instead of in the test (which would save you the `if`/`else`)

or if leaving logic as it is, should rename the field something like `date_last_active` and change its type annotation to `datetime.date | None`

either way, as long as words and logic agree